src/de/deserializer.zig

```diff
@@ -249,137 +267,104 @@ pub fn Deserializer(comptime user_dbt: anytype) type {
                             return error.InvalidType;
                         }
 
-                        return try visitor.visitInt(allocator, De, try std.fmt.parseInt(Value, slice, 10));
+                        return try visitor.visitInt(allocator, De, std.fmt.parseInt(Value, slice, 10) catch return error.InvalidType);
```

Comment:
The reason for this is because those are in contexts where a ParseIntError should only occur in case the JSON data contains a float, which is the wrong data type. It's true that I missed a few places where this also is the case but it's still using try.

Reply:
`ParseIntError` can also be returned by `std.fmt.parseInt` in the case of overflow (i.e., when the input data contains a number too large to store in `Value`). In which case, `InvalidType` wouldn't be the more accurate error to return.